rs/nns/governance/src/storage/neurons.rs

```diff
@@ -819,31 +872,47 @@ fn update_repeated_field<Element, Memory>(
     update_range(new_entries, range, map)
 }
 
+fn insert_element_in_repeated_field<Element, Memory>(
+    neuron_id: NeuronId,
+    index: u64,
+    element: Element,
+    map: &mut StableBTreeMap<(NeuronId, /* index */ u64), Element, Memory>,
+) where
+    Element: Storable + PartialEq,
+    Memory: ic_stable_structures::Memory,
+{
+    let key = (neuron_id, index);
+    map.insert(key, element);
+}
+
 /// Replaces the contents of map where keys are in range with new_entries.
 // TODO(NNS1-2513): To avoid the caller passing an incorrect range (e.g. too
 // small, or to big), derive range from NeuronId.
 fn update_range<Key, Value, Memory>(
-    new_entries: HeapBTreeMap<Key, Value>,
+    mut new_entries: HeapBTreeMap<Key, Value>,
```

Comment:
Note, changes to this function are generally an optimization, but did not end up being used in the final version of how to optimize vote recording.  But keeping them b/c they're better.